Improve test data generation

matti-lamppu opened the pull request:
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Improves test data generation for local development
    - More consistent naming
    - Add actual use cases
    - Add descriptions of reservation unit properties to its description
    - Add more descriptive reservation unit types for better searching 
    - Generate better seasonal bookings
    - Generate some recurring reservations
    - Make test data generation more configurable for later additions
    - Add reservation unit and purpose images
    - Fix reservation unit types for Caisa

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- None

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-3621](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3621)
- [TILA-3587](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3587)
- [TILA-3588](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3588)


[TILA-3621]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3621?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[TILA-3587]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3587?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

matti-lamppu commented:
@joonatank @vincit-matu Can you check this branch and check the generated test data for any needs? I haven't fixed the pictures yet, but the "bulk" data is now much different and I would like your input.

matti-lamppu commented:
Fixed caisa not being reservable due to not generating reservable time spans for it.

ranta reviewed (approved):
🚀🚀

matti-lamppu commented:
Added image fetching to test data for reservation units and purposes.

vergama commented on tilavarauspalvelu/management/commands/data_creation/create_reservation_units.py:
Is True for `allow_reservations_without_opening_hours` needed anymore if these have `origin_hauki_resource` assigned?

matti-lamppu replied:
Probably not

matti-lamppu commented:
Fixed reservation unit types for Caisa units

matti-lamppu commented:
Added recurring reservations (past, future and ongoing) with some rejected occurrences, denied and cancelled reservations. Also fixed pricing for non-fixed priced reservations

vergama commented on tilavarauspalvelu/management/commands/data_creation/create_caisa.py:
```diff
@@ -49,734 +54,686 @@ def _create_caisa(metadata_sets: dict[SetName, ReservationMetadataSet]) -> None:
-    tax_percentage_24, _ = TaxPercentage.objects.get_or_create(value=24.0)
-    payment_type_online, _ = ReservationUnitPaymentType.objects.get_or_create(code="ONLINE")
+    tax_percentage_0 = TaxPercentageFactory.create(value=0.0)
+    tax_percentage_24 = TaxPercentageFactory.create(value=24.0)
```
The tax percentage could be 25.5 now. (Matches test environment data, then.)

matti-lamppu replied:
Fixed.

matti-lamppu commented on tilavarauspalvelu/management/commands/data_creation/create_caisa.py:
```diff
@@ -789,186 +746,213 @@ def _create_caisa(metadata_sets: dict[SetName, ReservationMetadataSet]) -> None:
+        price_unit=PriceUnit.PRICE_UNIT_PER_HOUR,
+        lowest_price=0.0,
+        highest_price=40.0,
+        tax_percentage=tax_percentage_24,
```
Fixed.

ranta commented on tilavarauspalvelu/management/commands/data_creation/create_users.py:
```diff
@@ -1,19 +1,17 @@
     users: list[User] = [
         # Django admin user
-        User(
+        UserFactory.build(
```
These factories need to be updated, since `preferred_language` is now a non-nullable field after #1380

`django.db.utils.IntegrityError: null value in column "preferred_language" of relation "user" violates not-null constraint`

matti-lamppu replied:
Fixed

ranta commented on tilavarauspalvelu/management/commands/data_creation/create_reservation_units.py:
Missing the if check
```py
if image is not None:
    images.append(image)
````

matti-lamppu replied:
Good catch, fixed.

matti-lamppu commented:
/ff
